types/mongodb/index.d.ts

```diff
@@ -869,7 +869,7 @@ export type UpdateQuery<T> = {
     $currentDate?: { [P in keyof T]?: (true | { $type: 'date' | 'timestamp' }) } | { [key: string]: (true | { $type: 'date' | 'timestamp' }) };
     $addToSet?: Partial<T> | { [key: string]: any };
     $pop?: { [P in keyof T]?: -1 | 1 } | { [key: string]: -1 | 1 };
-    $pull?: Partial<T> | { [key: string]: Condition<T, keyof T> };
+    $pull?: Partial<T> | { [key: string]: Condition<T, keyof T> | any };
```

Comment:
This weakens `$pull` to `Partial<T> | { [key: string]: any }` as the union of `Condition<T, keyof T> | any` is reduced to `any`.

If you intend to weaken the type to `any`, I would suggest you remove `Condition<T, keyof T>` from the union.

Reply:
Also, can you provide a test case that you need working in the tests file?